workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/ModalScreenOverlayDialogFactory.kt

```diff
@@ -81,35 +79,23 @@ public open class ModalScreenOverlayDialogFactory<O : ScreenOverlay<*>>(
     initialRendering: O,
     initialEnvironment: ViewEnvironment,
     context: Context
-  ): Dialog {
-    // Put a no-op backPressedHandler behind the given rendering, to
-    // ensure that the `onBackPressed` call below will not leak up to handlers
-    // that should be blocked by this modal session.
-    val wrappedContentRendering = BackButtonScreen(initialRendering.content) { }
-
-    val contentViewHolder = wrappedContentRendering.toViewFactory(initialEnvironment)
-      .startShowing(wrappedContentRendering, initialEnvironment, context).apply {
-        // If the content view has no backPressedHandler, add a no-op one to
-        // ensure that the `onBackPressed` call below will not leak up to handlers
-        // that should be blocked by this modal session.
-        if (view.backPressedHandler == null) view.backPressedHandler = { }
+  ): OverlayDialogHolder<O> {
+    val contentViewHolder = initialRendering.content.toViewFactory(initialEnvironment)
+      .startShowing(initialRendering.content, initialEnvironment, context).apply {
+        environment[ModalScreenOverlayBackButtonHelper].initialize(view)
```

Comment:
Perhaps these should be methods on this interface, with default implementations?